### PR TITLE
ng_protonum: Initial import of protocol number defines

### PR DESCRIPTION
**Pull request by authmillenon**
Imports protocol numbers as they are needed for IPv4 and IPv6.

### CODE REVIEW

**Comment by OlegHahm** [comment]
Did you copied the list from somewhere or write it down yourself?

**Comment by authmillenon** [comment]
I generated it from the IANA's CSV file with a quick-hacked Python script and adapted the result to my liking (renamed e.g. `NG_PROTONUM_IPV6_ICMP` to `NG_PROTONUM_ICMPV6`

File: sys/include/net/ng_protnum.h [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_ng_protnum  Protocol Numbers
+ * @ingroup     net
+ * @brief       Defines for the Protocol Numbers as they are used in the
+ *              IPv4 protocol field and the IPv6 next header field
+ *              (ng_ipv6_hdr_t::nh).
```
Comment (LudwigOrtmann) [inline sys/include/net/ng_protnum.h]: Please include "Last Updated 2015-01-06"

Comment (authmillenon) [inline sys/include/net/ng_protnum.h]: Done (s. [l 18](https://github.com/RIOT-OS/RIOT/pull/2455/files#diff-23a386e4e5822f061414607685d0a239R18))

**Comment by OlegHahm** [comment]
@authmillenon:  Would you mind to share this script somehow so people might use it to update the list later?

**Comment by authmillenon** [comment]
It's not that sophisticated and requires a lot of extra work later on, to achieve the list I created:

```Python
import csv
import re

with open("protocol-numbers-1.csv") as csvfile:
    r = csv.DictReader(csvfile, delimiter=",")
    for row in r:
        if (len(row["Keyword"].strip()) == 0):
            continue
        ext = ""
        if row["IPv6 Extension Header"] == "Y":
            ext = "IPV6_EXT_"
        macro = "NG_PROTNUM_" + ext + re.sub("[^A-Z0-9_]", "_", row["Keyword"].upper()).strip("_")
        comment = row["Protocol"]

        if len(comment) == 0:
            comment = row["Keyword"]
        print("#define " + macro + ((32 - len(macro)) * " ") + ("(%3d)" % int(row["Decimal"])) + "   /**< " + comment + " */")
```

I think for updates it is just best to add the new values by hand. Or does the IANA updates this list in bulk?

**Comment by OlegHahm** [comment]
To be honest: no clue. But let's see...

**Comment by OlegHahm** [comment]
ACK

**Comment by OlegHahm** [comment]
but no beer for this one.

**Comment by LudwigOrtmann** [comment]
@authmillenon please squash

**Comment by authmillenon** [comment]
Done

**Pull request merged**